client/main.lua

```diff
@@ -1,274 +1,216 @@
-                }
+    local vehicleitems = {{
+        id = 'vehicle-flip',
+        label = Lang:t("options.flip"),
+        icon = 'car-burst',
+        onSelect = function()
+            TriggerEvent('radialmenu:flipVehicle')
+            lib.hideRadial()
+        end,
+    }}
+
+    vehicleitems[#vehicleitems+1] = convert(Config.VehicleDoors)
+    if Config.EnableExtraMenu then vehicleitems[#vehicleitems+1] = convert(Config.VehicleExtras) end
+
+    if Config.VehicleSeats then
+
+        CreateThread(function()
+            while true do
+                Wait(50)
+                if IsControlJustPressed(0, 23) and not IsPedInAnyVehicle(cache.ped, false) then
+                    local vehicle, _ = QBCore.Functions.GetClosestVehicle(GetEntityCoords(cache.ped))
+                    if vehicle then
+                        local vehicleseats = {}
+                        local seatTable = {
+                            [1] = Lang:t("options.driver_seat"),
+                            [2] = Lang:t("options.passenger_seat"),
+                            [3] = Lang:t("options.rear_left_seat"),
+                            [4] = Lang:t("options.rear_right_seat"),
+                        }
+
+                        local AmountOfSeats = GetVehicleModelNumberOfSeats(GetEntityModel(vehicle))
+                        for i = 1, AmountOfSeats do
+                            vehicleseats[#vehicleseats+1] = {
+                                id = 'vehicleseat'..i,
+                                label = seatTable[i] or Lang:t("options.other_seats"),
+                                icon = 'caret-up',
+                                onSelect = function()
+                                    if IsPedInAnyVehicle(cache.ped, false) then
+                                        TriggerEvent('radialmenu:client:ChangeSeat', i, seatTable[i] or Lang:t("options.other_seats"))
+                                    else
+                                        QBCore.Functions.Notify(Lang:t('error.not_in_vehicle'), 'error')
+                                    end
+                                    lib.hideRadial()
+                                end,
+                            }
+                        end
+                        lib.registerRadial({
+                            id = 'vehicleseatsmenu',
+                            items = vehicleseats
+                        })
+                    end
+                end
             end
-        end
+        end)
+        vehicleitems[#vehicleitems+1] = Config.VehicleSeats
     end
 
-    if #VehicleMenu.items == 0 then
-        if vehicleIndex then
-            RemoveOption(vehicleIndex)
-            vehicleIndex = nil
+    lib.registerRadial({
+        id = 'vehiclemenu',
+        items = vehicleitems
+    })
+    lib.addRadialItem(VehicleMenu)
+
+    for _, v in pairs(Config.MenuItems) do
+        lib.addRadialItem(convert(v))
+    end
+    if PlayerData.job.onduty then
```

Comment:
A guard clause here would be preferrable

Reply:
kinda hard to guard clause in that but i think it's ok now